doc/developers/advanced_installation.rst

```diff
@@ -22,127 +33,191 @@ basis to help users test bleeding edge features or bug fixes::
-can be done with::
+.. _git_repo:
 
-    pip install cython pytest flake8
+Use `Git <https://git-scm.com/>`_ to check out the latest source from the
+`scikit-learn repository <https://github.com/scikit-learn/scikit-learn>`_ on
+`Github <https://github.com>`_ and pip to build the project::
 
-Then, in the main repository::
+    git clone git://github.com/scikit-learn/scikit-learn.git
+    cd scikit-learn
 
+    pip install cython
     pip install --editable .
 
-Please read below for details and more advanced instructions.
+Check that the installed scikit-learn with a version number ending with
+`.dev0`::
+
+    python -c "import sklearn; sklearn.show_versions()"
+
+Run the tests with ``pytest``::
+
+    pip install pytest
+    pytest sklearn
+
+Please read below for more details and :ref:`platform system specific
+instructions <platform_specific_instructions>` (e.g. in case compilation error
+message).
+
+.. note::
+
+    You will have to re-run the ``pip install --editable .`` command every
+    time the source code of a compiled extension is changed (for instance when
+    switching branches or pulling changes from upstream). Compiled extensions
+    are Cython files (ending in `.pyx` or `.pxd`). See :ref:`editable_mode`
+    for more details.
 
 Dependencies
 ------------
 
-Scikit-learn requires:
+Runtime dependencies
+~~~~~~~~~~~~~~~~~~~~
+
+Scikit-learn requires the following dependencies both at build time and at
+runtime:
 
 - Python (>= 3.5),
 - NumPy (>= 1.11),
 - SciPy (>= 0.17),
 - Joblib (>= 0.11).
 
+Those dependencies are **automatically installed by pip and conda** when
+installing scikit-learn or building scikit-learn from source.
+
 .. note::
 
-   For installing on PyPy, PyPy3-v5.10+, Numpy 1.14.0+, and scipy 1.1.0+
+   For running on PyPy, PyPy3-v5.10+, Numpy 1.14.0+, and scipy 1.1.0+
    are required. For PyPy, only installation instructions with pip apply.
 
+Build dependencies
+~~~~~~~~~~~~~~~~~~
 
-Building Scikit-learn also requires
+Building Scikit-learn also requires:
 
-- Cython >=0.28.5
-- OpenMP
+- Cython >= 0.28.5
+- A C/C++ compiler and a matching `OpenMP
+  <https://en.wikipedia.org/wiki/OpenMP>`_ runtime library. See the
+  :ref:`platform system specific instructions
+  <platform_specific_instructions>` for more details.
 
 .. note::
 
    It is possible to build scikit-learn without OpenMP support by setting the
    ``SKLEARN_NO_OPENMP`` environment variable (before cythonization). This is
    not recommended since it will force some estimators to run in sequential
-   mode and their ``n_jobs`` parameter will be ignored.
+   mode.
+
+Since version 0.21, scikit-learn automatically detects and use the
+linear algebrea library used by SciPy **at runtime**. Scikit-learn has
+therefore no build dependency on BLAS/LAPACK implementations such as OpenBlas,
+Atlas or MKL.
 
+Test dependencies
+~~~~~~~~~~~~~~~~~
 
-Running tests requires
+Running tests requires:
 
 .. |PytestMinVersion| replace:: 3.3.0
```

Comment:
Actually there are bugs in `pytest` as low as 3.8.1 

@thomasjpfan recently updated the CI to 4.6.2 I think

Reply:
Fixed.